doc/deployment/openshift.md

```diff
@@ -1,112 +1,420 @@
+
+We're currently developing good rules of thumb for scaling this storage as your Pachyderm deployment grows,
+but it looks like 10G of disk space is sufficient for most purposes.
+
+### Object store
+
+Size your object store generously, once you start using Pachyderm, you'll start versioning all your data.
+You'll need four items to configure object storage
+
+1. The access endpoint.
+   For example, Minio's endpoints are usually something like `minio-server:9000`. 
+   Don't begin it with the protocol; it's an endpoint, not an url.
+1. The bucket name you're dedicating to Pachyderm. Pachyderm will need exclusive access to this bucket.
+1. The access key id for the object store.  This is like a user name for logging into the object store.
+1. The secret key for the object store.  This is like the above user's password.
+
+### TCP/IP ports
+
+For more details on how Kubernetes networking and service definitions work, see the [Kubernetes services 
+documentation](https://kubernetes.io/docs/concepts/services-networking/service/).
+
+#### Incoming ports (port)
+
+These are the ports internal to the containers, 
+You'll find these on both the pachd and dash containers.
+OpenShift runs containers and pods as unprivileged users which don't have access to port numbers below 1024.
+Pachyderm's default manifests use ports below 1024, so you'll have to modify the manifests to use other port numbers.
+It's usually as easy as adding a "1" in front of the port numbers we use.
+
+#### Pod ports (targetPort)
+
+This is the port exposed by the pod to Kubernetes, which is forwarded to the `port`.
+You should leave the `targetPort` set at `0` so it will match the `port` definition. 
+
+#### External ports (nodePorts)
+
+This is the port accessible from outside of Kubernetes.
+You probably don't need to change `nodePort` values unless your network security requirements or architecture requires you to change to another method of access. 
+Please see the [Kubernetes services documentation](https://kubernetes.io/docs/concepts/services-networking/service/) for details.
+
+## The OCPify script
+
+A bash script that automates many of the substitutions below is available [at this gist](https://gist.github.com/gabrielgrant/86c1a5b590ae3f4b3fd32d7e9d622dc8). 
+You can use it to modify a manifest created using the `--dry-run` flag to `pachctl deploy custom`, as detailed below, and then use this guide to ensure the modifications it makes are relevant to your OpenShift environment.
+It requires certain prerequisites, just as [jq](https://github.com/stedolan/jq) and [sponge, found in moreutils](https://joeyh.name/code/moreutils/).
+
+This script may be useful as a basis for automating redeploys of Pachyderm as needed. 
+
+### Configuration as code
+
+You're strongly encouraged to treat any manifests you generate and modify as _code_, subjecting them to the same revision control management you would any source code.
```

Comment:
Ideally, their configuration would actually _be_ code, rather than just being _treated_ as code. What I mean is the transformations themselves should be scripted so they can be automatically re-applied (as with the OCPify script). The resulting outputs should ideally also be versioned, but that's both kinda less important (so long as they always come from a repeatable process) and less useful, since the specific textual content of the manifests will generally change between runs of `pachctl deploy` (even though the semantic meaning will not) because lists of env var keys will be reordered (so diffs will be pretty noisy)